File: src/mobile/lib/app/utils/dto/notificacoes/notificacao_dto.dart
class NotificacaoDto {
  NotificacaoDto({
    required this.id,
    required this.titulo,
    required this.mensagem,
    this.imageUrl,
    required this.dataCriacao,
  });
  late final String id;
  late final String titulo;
  late final String mensagem;
  late final String? imageUrl;
  late final DateTime dataCriacao;

  NotificacaoDto.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    titulo = json['titulo'];
    mensagem = json['mensagem'];
    imageUrl = json['imageUrl'];
    dataCriacao = DateTime.parse(json['dataCriacao']);
  }
}

File: src/services/Catalogo/Catalogo.API/Data/Dto/NotificacaoDto.cs
using System.Text.Json;

namespace Catalogo.API.Data.Dto
{
  public record NotificacaoDto
  {
    public string Id { get; set; } = null!;

    public string Titulo { get; set; } = null!;

    public string Mensagem { get; set; } = null!;

    public string? ImageUrl { get; set; }

    public DateTimeOffset DataCriacao { get; set; } = DateTimeOffset.UtcNow;

    public override string ToString()
    {
      return JsonSerializer.Serialize(this);
    }
  }
}